### PR TITLE
fix(ui-bug): Fix mission waypoint/stopover/mark pointers disappearing

### PR DESCRIPTION
**Pull request by warp-core**
**Bug fix**

This PR addresses the bug/feature described in issue #10840

## Summary
When drawing mission pointers on the map, they are drawn mission by mission (for active missions).
There is a check on if the destination system of the mission currently under consideration already has the maximum number of pointers allowed for active missions (some spaces my be reserved for available jobs).
If the destination already has enough pointers, drawing the waypoint, stopover, and mark pointers for this mission on other systems are also skipped, resulting in the behaviour where the waypoint pointer on Mimosa is not drawn because the destination pointer for that mission on Aludra is not being drawn because that system already has the maximum number of pointers.
This PR makes it so that pointers on other systems are not skipped even for missions where the destination pointer is skipped.

## Screenshots
before | after
-- | --
image | image

## Usage examples
N/A

## Testing Done
Checked the map with the attached save file.

## Save File
This save file can be used to test these changes:
[Lakna Byrdock mission marker testing.txt](https://github.com/user-attachments/files/18045874/Lakna.Byrdock.mission.marker.testing.txt)
Based on the save file provided in the linked issue by tekker2234.
Ignore the disabled plugin mission.

## Artwork Checklist
N/A

## Wiki Update
N/A

## Performance Impact
`MapPanel::DrawMissions` is marginally slower; on my system, the Release build from clang 15 is about 10 microseconds slower, out of around 260-270 microseconds.

### CODE REVIEW

File: source/MapPanel.cpp [hunk 1]
```diff
@@ -1463,39 +1478,44 @@ void MapPanel::DrawMissions()
...
-
-		pair<bool, bool> blink = BlinkMissionIndicator(player, mission, step);
-		bool isSatisfied = IsSatisfied(player, mission) && blink.second;
-		DrawPointer(system, it.drawn, blink.first ? black : isSatisfied ? currentColor : blockedColor, isSatisfied);
+		auto &it = missionCount[system];
+		if(it.drawn < it.MaximumActive())
+		{
+			pair<bool, bool> blink = BlinkMissionIndicator(player, mission, step);
+			bool isSatisfied = IsSatisfied(player, mission) && blink.second;
+			const Color &color = blink.first ? black : isSatisfied ? currentColor : blockedColor;
```
Comment (warp-core) [inline source/MapPanel.cpp:1486]:
```suggestion
			const Color &color = (blink.first ? black : isSatisfied ? currentColor : blockedColor);
```

**Comment by tekker2234** [comment]
apologies for the plugin message, I didn't realize I had some enabled when I tested this.
Here is one that shouldn't have the warning.
[Lakna Byrdock~Mission Marker Testing.txt](https://github.com/user-attachments/files/18045967/Lakna.Byrdock.Mission.Marker.Testing.txt)

File: source/MapPanel.cpp [hunk 2]
```diff
@@ -69,13 +69,27 @@ namespace {
...
+	class PointerDrawCount {
+	public:
+		// Calculate and check the most number of pointer positions that should be available for active missions.
+		// This can be up to half the maximum number of pointers that can be drawn.
+		void Reserve()
+		{
+			maximumActive = max(MAX_MISSION_POINTERS_DRAWN / 2, MAX_MISSION_POINTERS_DRAWN - (available + unavailable));
+		}
+
+		unsigned MaximumActive() const { return maximumActive; }
```
Comment (TomGoodIdea) [inline source/MapPanel.cpp:83]: There's the style rule that says no function bodies inside the class body, but there are other cases of this in the current code, so we can leave it for another PR.

Comment (tibetiroka) [reply source/MapPanel.cpp:83]: It still has to be on a separate line in this case, right?
```suggestion
		void Reserve()
		{
			maximumActive = max(MAX_MISSION_POINTERS_DRAWN / 2, MAX_MISSION_POINTERS_DRAWN - (available + unavailable));
		}

		unsigned MaximumActive() const
		{
			return maximumActive;
		}
```

Also, is there any reason to make this into a class if we just put public fields in it?

Comment (warp-core) [reply source/MapPanel.cpp]: I've moved the definitions out of the class.

> Also, is there any reason to make this into a class if we just put public fields in it?

It has one private field and it has a method handling it. ~~We also just shouldn't use the "struct" keyword per the style guide.~~

Comment (petervdmeer) [reply source/MapPanel.cpp]: Those review-comments appear to be handled.

File: source/MapPanel.cpp (hunk 1 above)
Comment (petervdmeer) [reply source/MapPanel.cpp:1494]: I think it works either way. (And you are the PR author, so if you feel that this change is needed, then just apply it.)

**Review by petervdmeer — APPROVED** [review]
